river/Seat.zig

```diff
@@ -350,29 +359,42 @@ pub fn handleMapping(
                     log.err("failed to update mapping repeat timer", .{});
                 };
             }
-            self.runMappedCommand(mapping);
+            self.runCommand(mapping.command_args);
             return true;
         }
     }
     return false;
 }
 
-fn runMappedCommand(self: *Self, mapping: *const Mapping) void {
+/// Handle any user-defined mapping for switches
+pub fn handleSwitchMapping(
+    self: *Self,
+    switch_type: Switch.Type,
+    switch_state: Switch.State,
+) void {
+    const modes = &server.config.modes;
+    for (modes.items[self.mode_id].switch_mappings.items) |mapping| {
+        if (std.meta.eql(mapping.switch_type, switch_type) and std.meta.eql(mapping.switch_state, switch_state)) {
+            self.runCommand(mapping.command_args);
+        }
+    }
+}
+
+fn runCommand(self: *Self, command_args: []const [:0]const u8) void {
```

Comment:
the diff would be cleaner if we kept this variable as just `args`.
```suggestion
fn runCommand(self: *Self, args: []const [:0]const u8) void {
```